### PR TITLE
[proofs] [alethe] Update doc, remove experimental option

### PR DESCRIPTION
**Pull request by HanielB**

### CODE REVIEW

**Review by ajreynol — CHANGES_REQUESTED** [review]
This is pending experiments.

**Comment by HanielB** [comment]
What experiments do you have in mind?

**Comment by ajreynol** [comment]
Running some sanity checks over SMT-LIB for the printer now, it would be good to have a apples-to-apples comparison of CPC+Ethos vs. Alethe+Carcara on all logics fully supported by Alethe+Carcara.

**Comment by HanielB** [comment]
I'll kick off an SMT-LIB run in AUFBVLIRADT (or something like that) that should include what is supported by cvc5+Carcara, it'll be a good sanity check indeed. Comparing with CPC+Ethos I think would be nice also but less priority, so I'd rather leave for later. What do you think?

**Comment by ajreynol** [comment]
That's fine but it might make it harder to interpret the results, what would be ideal is to know what is the overhead of using Alethe vs. CPC.

**Comment by HanielB** [comment]
I agree it would be nice to compare like this, but I don't think this should be a criterion to remove the experimental flag.

File: docs/proofs/output_alethe.rst [hunk 1]
```diff
@@ -10,16 +10,17 @@ SMT-LIB.  It includes both coarse- and fine-grained steps and was first
...
 as well as within Coq, the latter via the SMTCoq plugin :cite:`Armand2011,
-Ekici2017`. There is also a high performance Rust proof checker for Alethe,
-available `here <https://github.com/ufmg-smite/carcara>`_. For a quick start,
-the cvc5 repository contains a :cvc5repo:`script <contrib/get-carcara-checker>`
-to download and install the Carcara checker's version with full support for cvc5
+Ekici2017`. There is also a high performance Rust proof checker and elaborator
+for Alethe: Carcara, available `here
+<https://github.com/ufmg-smite/carcara>`_. For a quick start, the cvc5
+repository contains a :cvc5repo:`script <contrib/get-carcara-checker>` to
+download and install the Carcara checker's version with full support for cvc5
```
[the remark this answers is not in the document]
Comment (HanielB) [reply docs/proofs/output_alethe.rst:17]: It is not HEAD, unfortunately. I have to merge a number of updates (for bit-vectors and datatypes, mostly) into the main of Carcara and have not gotten the time yet.

**Comment by HanielB** [comment]
Running on SMT-LIB on logics (with or without quantifiers) with AUFBVDTLIRA, 300s, 8gb, comparing against `main` without proofs, there are no disagreements and no more errors, the impact is only in performance:
```
                       status  total  solved    sat  unsat   best  timeout  memout  error  uniq  disagr  time_cpu     memory
benchmark   config                                                                                                          
ABV         main/          ok   4975     107     17     90    107        3       0      0     0       0     908.3    14820.6
            pf-alethe/     ok   4975     108     17     91     81        7       0      0     1       0    2461.4    43835.7
ALIA        main/          ok   3098     108     16     92    104        3       0      0     0       0     936.8     9889.8
            pf-alethe/     ok   3098     108     16     92     66        3       0      0     0       0     956.1    13091.1
AUFBV       main/          ee   1523     190      0    190    188      154       4    156    47       0   63898.7   424650.3
            pf-alethe/     ee   1523     147      0    147      6      278      81    156     4       0  121845.9  1821400.0
AUFBVDTLIA  main/          ok   1683     392      4    388    392      701       3      0     4       0  210953.6   413324.0
            pf-alethe/     ok   1683     390      4    386    296      696      10      0     2       0  211511.5  1039604.3
AUFDTLIA    main/          ok    795     697      3    694    697       35       0      0     0       0   10570.4    17333.8
            pf-alethe/     ok    795     697      3    694    614       35       0      0     0       0   11146.1    31790.9
AUFDTLIRA   main/          ok  11043    9904      0   9904   9900       59       0      0     1       0   18798.3    60003.1
            pf-alethe/     ok  11043    9903      0   9903   6025       60       0      0     0       0   19515.2   143978.2
AUFLIA      main/          ok   3327    2037     28   2009   2036      421       3      0     0       0  130627.0   358107.9
            pf-alethe/     ok   3327    2037     28   2009   1579      400      24      0     0       0  128818.0   816289.2
AUFLIRA     main/          ok  20011   19743      0  19743  19741      112       0      0     0       0   36607.7    90725.0
            pf-alethe/     ok  20011   19743      0  19743  17467      125     192      0     0       0   80192.1  2454637.7
BV          main/          ok   6185    5651    446   5205   5634      534       0      0   484       0  166605.7   291634.8
            pf-alethe/     ok   6185    5169    321   4848   4051     1008      48      0     2       0  321978.0  2440769.5
LIA         main/          ok    416     416    150    266    415        0       0      0     0       0      78.5     1341.9
            pf-alethe/     ok    416     416    150    266    258        0       0      0     0       0     102.4     2666.4
LRA         main/          ok   2439    1915    597   1318   1911      524       0      0     1       0  162871.9    64288.9
            pf-alethe/     ok   2439    1914    596   1318    994      528       0      0     0       0  163801.1   159013.7
QF_ABV      main/          ok  15148   14429   9922   4507  13275      702      17      0   272       0  298389.2   922279.0
            pf-alethe/     ok  15148   14263   9866   4397   8646      821      98      0   106       0  356058.9  3867198.0
QF_ALIA     main/          ok    176     111     40     71     95       65       0      0     6       0   23418.9    40026.5
            pf-alethe/     ok    176     105     35     70     39       71       0      0     0       0   24354.5    73997.1
QF_ANIA     main/          ok    157     117    103     14    108       40       0      0     2       0   14626.5    15234.1
            pf-alethe/     ok    157     116    103     13     76       42       0      0     1       0   15883.7    38589.4
QF_AUFBV    main/          ok     75      43     12     31     43       22      10      0     2       0    7877.0   124768.2
            pf-alethe/     ok     75      42     11     31      4       19      15      0     1       0    7936.5   180401.0
QF_AUFLIA   main/          ok   1303    1303    543    760   1252        0       0      0     3       0    2006.9    13284.2
            pf-alethe/     ok   1303    1300    543    757    839        9       0      1     0       0    6419.2   120020.3
QF_AX       main/          ok    551     551    272    279    551        0       0      0     2       0     983.0     2903.9
            pf-alethe/     ok    551     549    272    277    275        2       0      0     0       0    2583.8    69226.0
QF_BV       main/          ok  46191   43338  17207  26131  43016     2756      97      0  7059       0 1170330.7  5482827.1
            pf-alethe/     ok  46191   36387  14103  22284  14117     9696    2197      0   108       0 3207987.4 47494760.4
QF_DT       main/          ok   8700    8253   3648   4605   8243      447       0      0    11       0  142534.4   254007.9
            pf-alethe/     ok   8700    8246   3647   4599   8063      466       0      0     4       0  147974.8  2401720.6
QF_IDL      main/          ok   2528    1597    902    695   1571      921      10      0   183       0  325240.4   865592.9
            pf-alethe/     ok   2528    1420    788    632    194     1103      75      0     6       0  384326.6  2440207.8
QF_LIA      main/          ok  13306    9784   6822   2962   9566     3460      62      0   271       0 1149785.6  4189711.4
            pf-alethe/     ok  13306    9552   6733   2819   6503     3876     116      0    39       0 1253326.8  8094056.9
QF_LIRA     main/          ok      7       5      1      4      5        2       0      0     0       0     628.1     1418.9
            pf-alethe/     ok      7       5      1      4      0        2       1      0     0       0     970.1    21391.9
QF_LRA      main/          ok   1753    1514    890    624   1474      239       0      0    37       0  108611.6   206982.1
            pf-alethe/     ok   1753    1487    878    609    654      289       1      0    10       0  130425.8   727045.5
QF_RDL      main/          ok    255     202    101    101    200       53       0      0     5       0   20785.0    21207.7
            pf-alethe/     ok    255     197    101     96     13       62       0      0     0       0   25639.5   130780.7
QF_UF       main/          ok   7503    7493   3142   4351   7395       10       0      0     8       0    9066.6    56082.6
            pf-alethe/     ok   7503    7485   3142   4343   1932       26       1      0     0       0   32173.0   598466.6
QF_UFBV     main/          ok   1510    1318    710    608   1229      127      65      0    54       0   59079.3   938715.2
            pf-alethe/     ok   1510    1287    683    604    179      121     131      0    23       0   68372.2  1744452.2
QF_UFBVDT   main/          ok     76      33     28      5     33       14      29      0    16       0    8044.1   328275.4
            pf-alethe/     ok     76      17     15      2      0        5      54      0     0       0    7432.6   513035.7
QF_UFDT     main/          ok    203     101     78     23     85      102       0      0    15       0   41008.7    62322.1
            pf-alethe/     ok    203      88     77     11     26      117       0      0     2       0   43200.7   207672.9
QF_UFDTLIA  main/          ok     76      30     23      7     23       46       0      0     1       0   14811.3    30306.5
            pf-alethe/     ok     76      34     28      6     13       41       1      0     5       0   14691.7    71501.5
QF_UFDTLIRA main/          ok    146     146     80     66    146        0       0      0     0       0       0.0      281.7
            pf-alethe/     ok    146     146     80     66    146        0       0      0     0       0       0.0      372.9
QF_UFIDL    main/          ok    628     412    134    278    405      216       0      0    18       0   77835.4    91065.8
            pf-alethe/     ok    628     397    134    263     56      241      36      0     3       0   89474.4   858405.1
QF_UFLIA    main/          ok    659     615    426    189    565       44       0      0     2       0   16168.9    44780.2
            pf-alethe/     ok    659     613    425    188    315       45       2      0     0       0   17620.0   123241.3
QF_UFLRA    main/          ok   1284    1276    765    511   1257        7       1      0     9       0    4225.9    95933.9
            pf-alethe/     ok   1284    1267    759    508    606        0      26      0     0       0    2454.5   294941.1
QF_UFNRA    main/          ok     58      40     16     24     29       18       0      0     8       0    6919.3     2236.8
            pf-alethe/     ok     58      33      8     25     25       25       0      0     1       0    7582.5     5100.9
UF          main/          ok   7590    3163      1   3162   3156     3706       0      0     1       0 1119699.4  1036914.0
            pf-alethe/     ok   7590    3163      1   3162   2094     3703       3      0     1       0 1120194.0  2069497.4
UFBV        main/          ok    193      60      8     52     60        3       1      0     1       0    1729.8    27657.2
            pf-alethe/     ok    193      59      8     51     23        3       2      0     0       0    2059.3    45548.2
UFBVDT      main/          ok      1       0      0      0      0        0       0      0     0       0      14.9      571.2
            pf-alethe/     ok      1       0      0      0      0        0       0      0     0       0      38.6     2744.6
UFBVLIA     main/          ok    208       0      0      0      0        0       0      0     0       0     189.8    12832.8
            pf-alethe/     ok    208       0      0      0      0        0       0      0     0       0     190.9    15316.8
UFDT        main/          ok   4569    2163      0   2163   2155     1985       0      1     0       0  601401.7   795583.1
            pf-alethe/     ok   4569    2164      0   2164   1713     1984       0      1     1       0  601447.9  1494807.1
UFDTLIA     main/          ok    328      77      0     77     77      239       0      0     0       0   71932.9   367515.8
            pf-alethe/     ok    328      77      0     77     57      239       0      0     0       0   71961.6   451251.4
UFDTLIRA    main/          ok   7749    5935    100   5835   5933        6       0      0     0       0    1921.8    32489.9
            pf-alethe/     ok   7749    5935    100   5835   4861        6       0      0     0       0    2053.4    44769.6
UFIDL       main/          ok     68      56      0     56     56        5       0      0     0       0    1504.0     2396.1
            pf-alethe/     ok     68      56      0     56     34        5       0      0     0       0    1506.4     3072.5
UFLIA       main/          ok  10128    7435      0   7435   7421     1934       0      1     6       0  596064.0   747405.9
            pf-alethe/     ok  10128    7429      0   7429   2998     1939       3      1     0       0  599846.3  1448598.6
UFLRA       main/          ok     15      10      0     10     10        0       0      0     0       0       0.0       31.2
            pf-alethe/     ok     15      10      0     10     10        0       0      0     0       0       0.0       44.4
```

Next I'm gonna do the sanity check of running Carcara on all the Alethe proofs generated in this run.

**Comment by ajreynol** [comment]
This looks good to me, @hansjoergschurr did you have an open question to address?

**Review by ajreynol — APPROVED** [review]

File: docs/proofs/output_alethe.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (HanielB) [reply docs/proofs/output_alethe.rst:17]: I think it can wait, since we are using the script and this is explained. But I'll try to address this ASAP.

**Pull request merged**